pyfar/dsp/dsp.py

```diff
@@ -1498,3 +1499,138 @@ def decibel(signal, domain='freq', log_prefix=None, log_reference=1,
+    Returns
+    --------
+    averaged_signal: Signal
+        averaged input Signal
+
+    Notes
+    -----
+    The functions :py:func:`~pyfar.dsp.linear_phase` and
+    :py:func:`~pyfar.dsp.minimum_phase` can be used to obtain non-zero phase
+    responses. This can be usefull if the average mode discards the phase.
+    """
+
+    # check input
+    if not isinstance(signal, (pyfar.Signal, pyfar.FrequencyData,
+                               pyfar.TimeData)):
+        raise TypeError(("Input data has to be of type 'Signal', 'TimeData' "
+                         "or 'FrequencyData'."))
+    if type(signal) == pyfar.FrequencyData and mode == 'time':
+        raise ValueError((
+            f"mode is '{mode}' and signal is type '{signal.__class__}'"
+            " but must be of type 'Signal' or 'TimeData'."))
+    if type(signal) == pyfar.TimeData and mode in ('log_magnitude_zerophase',
+                                                   'magnitude_zerophase',
+                                                   'magnitude_phase', 'power',
+                                                   'complex'):
+        raise ValueError((
+            f"mode is '{mode}' and signal is type '{signal.__class__}'"
+            " but must be of type 'Signal' or 'FrequencyData'."))
+    # check for axis
+    if axis and np.max(axis) > len(signal.cshape):
+        raise ValueError('The maximum of axis needs to be smaller then '
+                         'len(signal.cshape).')
+
+    # set axis default
+    if axis is None:
+        axis = tuple([i for i in range(len((signal.cshape)))])
+    if 1 in signal.cshape:
+        for i, ax in enumerate(axis):
+            if signal.cshape[ax] == 1:
+                warnings.warn("Sinnvolles Warning")
+    # set weights default
+    if weights is None:
+        weights = 1
+        for idx, size in enumerate(signal.cshape):
+            if idx in axis:
+                weights /= signal.cshape[idx]
+    # apply weights
+    signal *= weights
+
+    if not isinstance(axis, int):
+        axis = tuple([ax-1 if ax < 0 else ax for ax in axis])
+    # convert data to desired domain
+    if mode == 'time':
+        data = signal.time
+    elif mode == 'complex':
+        data = signal.freq
+    elif mode == 'magnitude_zerophase':
+        data = np.abs(signal.freq)
+    elif mode == 'magnitude_phase':
+        data = np.abs(signal.freq) * \
+               np.exp(1j*pyfar.dsp.phase(signal, unwrap=True))
+    elif mode == 'power':
+        data = np.abs(signal.freq)**2
+    elif mode == 'log_magnitude_zerophase':
+        data, log_prefix = pyfar.dsp.decibel(signal, 'freq',
+                                             prefix_return=True)
+    else:
+        raise ValueError(
+            """mode must be 'time', 'complex', 'magnitude_zerophase', 'power',
+            'magnitude_phase' or 'log_magnitude_zerophase'."""
+            )
+    # average the data
+    data = np.sum(data, axis=axis, keepdims=keepdims)
```

Comment:
You can also use np.average, which supports weighted averaging. 
https://numpy.org/doc/stable/reference/generated/numpy.average.html
This way you can also avoid the looped version of calculating the weights.